pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -1905,10 +1987,48 @@ def coalesce_object(self, ctx, auth_cb):
                         return denial_response
                 authed_containers.add(elem_container_path)
 
+        # treat the number of objects as the number of writes
```

Comment:
well, yes and no!

formerly (before this patch) proxyfs tracked the number of writes for each object in the SLO and then added them together and returned that number to `parse_coalesce_object_response()`, which means it could have been different from the number of objects.

this patch also changes the proxyfs code to return (and store) the number of objects as the number of writes, instead of the total number of writes to create the component objects.

since the result of the COALESCE is a new object, i think we can assign any number of writes we want (including "1") and the ETag code will still work correctly.  so i think i could have just made it 1.

i changed the way num_writes is calculated by coalesce to try and make the whole operation more transactional.  

if i don't know `num_writes` until `parse_coalesce_object_response()` returns, then i can't mung the metadata until then and i would have to perform a second transaction to set the correct metadata after the coalesce returns, which opens the door (or window) to it never getting done (if the node crashes between updates).

so i took a sort of "middle of the road" approach my using the number of objects as the number of writes instead of the actual number of writes or simply using 1.

Reply:
i updated the comment to:
```
        # proxyfs treats the number of objects as the number of writes                                                   
```